packages/palette/src/elements/PhoneInput/PhoneInput.tsx

```diff
@@ -0,0 +1,501 @@
+        className={className}
+        {...boxProps}
+      >
+        <ContainerBox
+          ref={anchorRef as any}
+          isDropdownVisible={isDropdownVisible}
+          hover={hover}
+          focus={focus}
+          error={error}
+          disabled={disabled}
+          placeholder={inputProps.placeholder}
+        >
+          <SelectContainer
+            ref={countryPickerRef as any}
+            disabled={disabled}
+            onClick={() => {
+              if (!disabled && !isDropdownVisible) {
+                setDropdownVisible(true)
+              } else {
+                setDropdownVisible(false)
+              }
+            }}
+            tabIndex={disabled ? -1 : 0}
+            onKeyDown={handleCountryPickerKeydown}
+          >
+            {selectedOption.text}
+          </SelectContainer>
+
+          <StyledInput
+            disabled={disabled}
+            ref={composeRefs(inputRef, forwardedRef) as any}
+            type="tel"
+            autoComplete="tel-national"
+            name={inputName}
+            maxLength={25}
+            placeholder={inputProps.placeholder}
+            onChange={(e) => {
+              const inputValue = e.target.value
+              const countryCodeValue = selectedOption.value
+              inputProps.onChange?.(e)
+
+              onInputChange?.({ inputValue, countryCodeValue })
+            }}
+            {...inputProps}
+          />
+
+          <StyledLabel htmlFor={inputName}>Phone Number</StyledLabel>
+        </ContainerBox>
+
+        {isDropdownVisible && (
+          <SelectDropdown ref={tooltipRef as any} role="listbox" width={width}>
+            <Input
+              ref={searchInputRef}
+              mb={1}
+              autoFocus
+              placeholder="Search"
+              onChange={(e) => setSearchQuery(e.target.value)}
+              onKeyDown={handleSearchInputKeydown}
+            />
+
+            <SelectOptions>
+              {optionsWithRefs.map(({ option, ref }, i) => {
+                return (
+                  <SelectOption
+                    key={i}
+                    ref={ref as any}
+                    role="option"
+                    aria-selected={option.value === selectedOption.value}
+                    aria-posinset={i + 1}
+                    aria-setsize={options.length}
+                    selected={option.value === selectedOption.value}
+                    onClick={() => handleSelect(option)}
+                    tabIndex={-1}
+                  >
+                    <Text minWidth={80}>{option.text}</Text>
+                    <Text>{option.name}</Text>
+                  </SelectOption>
+                )
+              })}
+            </SelectOptions>
+          </SelectDropdown>
+        )}
+
+        {required && !(error && typeof error === "string") && (
+          <Text mt={0.5} ml={1} variant="xs" color="black60" textAlign="left">
+            *Required
+          </Text>
+        )}
```

Comment:
Is there a *Required component we can reuse?

Reply:
No we don't but that's a good idea :) I'll add it now!